collections/src/main/scala/strawman/collection/DefaultMap.scala

```diff
@@ -0,0 +1,37 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___     Scala API                            **
+**    / __/ __// _ | / /  / _ |    (c) 2003-2013, LAMP/EPFL             **
+**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
+** /____/\___/_/ |_/____/_/ | |                                         **
+**                          |/                                          **
+\*                                                                      */
+
+package strawman
+package collection
+
+/** A default map which builds a default `immutable.Map` implementation for all
+  * transformations.
+  *
+  *  Instances that inherit from `DefaultMap[K, V]` still have to define:
+  *  {{{
+  *    def get(key: K): Option[V]
+  *    def iterator(): Iterator[(K, V)]
+  *  }}}
+  *
+  *  It might also be advisable to override `foreach` or `size` if efficient
+  *  implementations can be found.
+  *
+  *  @since 2.8
+  */
+trait DefaultMap[K, +V] extends Map[K, V] { self =>
+
+  // Members declared in IterableOps
+  def iterableFactory: IterableFactoryLike[Iterable] = Iterable
```

Comment:
You could even refine the return type to `IterableFactory[Iterable]`

Reply:
I wonder if we need `IterableFactoryLike` at all. The only use is in `View` where automatic buffering would be a sensible alternative with a simpler API.

Reply:
We could try something like that, indeed.